project-base/storefront/components/Pages/Customer/customerChangeProfileFormMeta.ts

```diff
@@ -29,16 +29,8 @@ export const useCustomerChangeProfileForm = (
     const resolver = yupResolver(
         Yup.object().shape<Record<keyof CustomerChangeProfileFormType, any>>({
             email: validateEmail(t),
-            passwordFirst: Yup.string().when('passwordOld', {
-                is: (passwordOld: string) => passwordOld.length > 0,
-                then: validateFirstPassword(t),
-                otherwise: Yup.string(),
-            }),
-            passwordSecond: Yup.string().when('passwordFirst', {
-                is: (passwordFirst: string) => passwordFirst.length > 0,
-                then: validateSecondPassword(t),
-                otherwise: Yup.string(),
-            }),
+            passwordFirst: validatePassword(t),
+            passwordSecond: validatePasswordAgain(t),
```

Comment:
This change has broken the current logic of customer edit form. The logic before was that if the old password was not provided, it was assumed that the user is not trying to change his password and passwords were therefore not validated. Your change results in passwords being always validated.

Reply:
In this video (your state) you can see the unnecessary/unwanted validation

https://github.com/shopsys/shopsys/assets/55181354/3c8ce015-5084-4017-ae9b-002cd8bcdaba

Reply:
In this video (previous state), you can see that if I did not provide old password, new ones were not validated


https://github.com/shopsys/shopsys/assets/55181354/64cf03c1-0a54-496c-92b2-4cb8c055f94e